### PR TITLE
Replace assert calls by simpler but equivalent calls.

### PR DESCRIPTION
**Pull request by arturobernalg**

### CODE REVIEW

File: httpclient5/src/test/java/org/apache/hc/client5/http/auth/TestCredentials.java [hunk 1]
```diff
@@ -84,9 +84,9 @@ public void testUsernamePasswordCredentialsHashCode() {
         final UsernamePasswordCredentials creds3 = new UsernamePasswordCredentials(
                 "name", "otherpwd".toCharArray());
 
-        Assert.assertTrue(creds1.hashCode() == creds1.hashCode());
+        Assert.assertEquals(creds1.hashCode(), creds1.hashCode());
         Assert.assertTrue(creds1.hashCode() != creds2.hashCode());
```
Comment (michael-o) [inline httpclient5/src/test/java/org/apache/hc/client5/http/auth/TestCredentials.java:88]: Why didn#t you modify this one?

File: httpclient5/src/test/java/org/apache/hc/client5/http/auth/TestCredentials.java [hunk 2]
```diff
@@ -124,15 +124,15 @@ public void testNTCredentialsHashCode() {
         final NTCredentials creds9 = new NTCredentials(
                 "name","pwd".toCharArray(), "somehost", null);
 
-        Assert.assertTrue(creds1.hashCode() == creds1.hashCode());
+        Assert.assertEquals(creds1.hashCode(), creds1.hashCode());
         Assert.assertTrue(creds1.hashCode() != creds2.hashCode());
```
Comment (michael-o) [inline httpclient5/src/test/java/org/apache/hc/client5/http/auth/TestCredentials.java:128]: Same here

**Review by michael-o — CHANGES_REQUESTED** [review]

File: httpclient5/src/test/java/org/apache/hc/client5/http/auth/TestCredentials.java (hunk 1 above)
Comment (arturobernalg) [reply httpclient5/src/test/java/org/apache/hc/client5/http/auth/TestCredentials.java:88]: I think it's more clear. There is a specific assertions for compare the two object

Comment (michael-o) [reply httpclient5/src/test/java/org/apache/hc/client5/http/auth/TestCredentials.java:88]: `assertNotEquals()` isn't clear?

Comment (arturobernalg) [reply httpclient5/src/test/java/org/apache/hc/client5/http/auth/TestCredentials.java:88]: Ok, reverted.
TY

File: httpclient5/src/test/java/org/apache/hc/client5/http/auth/TestCredentials.java (hunk 2 above)
Comment (arturobernalg) [reply httpclient5/src/test/java/org/apache/hc/client5/http/auth/TestCredentials.java:128]: Ok, reverted.
TY

File: httpclient5/src/test/java/org/apache/hc/client5/http/auth/TestCredentials.java (hunk 1 above)
Comment (michael-o) [reply httpclient5/src/test/java/org/apache/hc/client5/http/auth/TestCredentials.java:88]: I am confused now. I have reverted the first line but left the third one? It is now even more inconsistent. My question was: Why can the second line use `assertNotEquals()`?

Comment (arturobernalg) [reply httpclient5/src/test/java/org/apache/hc/client5/http/auth/TestCredentials.java]: Sorry. My mistake. I leave it as it was

**Comment by ok2c** [comment]
@michael-o Is it OK to merge this PR?

**Comment by michael-o** [comment]
> 
> 
> @michael-o Is it OK to merge this PR?

Yes, go ahead. Further improvements can be done in separate PRs!

**Pull request merged**